packages/admin/cms-admin/src/common/MasterMenu.tsx

```diff
@@ -2,16 +2,20 @@ import { Menu, MenuCollapsibleItem, MenuContext, MenuItemRouterLink, MenuItemRou
 import * as React from "react";
 import { RouteProps, useRouteMatch } from "react-router-dom";
 
+import { CurrentUserContext } from "../userPermissions/hooks/currentUser";
+
 export type MasterMenuItem = Omit<MenuItemRouterLinkProps, "to"> & {
+    requiredPermission?: string;
     route?: RouteProps;
     to?: string;
     submenu?: MasterMenuItem[];
 };
 
 export type MasterMenuData = MasterMenuItem[];
 
-export function getMenuFromMasterMenuData(items: MasterMenuData): MenuItem[] {
-    // TODO: Filter for user-permissions once they are available
+export function useMenuFromMasterMenuData(items: MasterMenuData): MenuItem[] {
+    const context = React.useContext<CurrentUserContext | undefined>(CurrentUserContext);
```

Comment:
the type can be inferred: `const context = React.useContext(CurrentUserContext);`

Reply:
76f3e2bba151574e003e9db0da4e0733f7974c90